stdlib/test/bit/test_bit.mojo

```diff
@@ -19,95 +19,378 @@ from bit import (
     bit_ceil,
     bit_floor,
     has_single_bit,
+    countl_zero,
+    countr_zero,
+    bit_reverse,
+    byte_swap,
+    pop_count,
+    bit_not,
 )
 
 from testing import assert_equal
 
 
+def test_countl_zero():
+    assert_equal(countl_zero(-(2**59)), 0)
+    assert_equal(countl_zero(-(2**20)), 0)
+    assert_equal(countl_zero(-1), 0)
+    assert_equal(countl_zero(-1), 0)
+    assert_equal(countl_zero(0), 64)
+    assert_equal(countl_zero(1), 63)
+    assert_equal(countl_zero(2), 62)
+    assert_equal(countl_zero(3), 62)
+    assert_equal(countl_zero(4), 61)
+    assert_equal(countl_zero(2**20), 43)
+    assert_equal(countl_zero(2**59), 4)
+
+
+def test_countl_zero_simd():
+    alias simd_width = 4
+    alias int8_t = DType.int8
+    alias int16_t = DType.int16
+    alias int32_t = DType.int32
+    alias int64_t = DType.int64
+
+    alias var1 = SIMD[int8_t, simd_width](-(2**6), 0, -1, 2**6)
+    assert_equal(countl_zero(var1), SIMD[int8_t, simd_width](0, 8, 0, 1))
+
+    alias var3 = SIMD[int16_t, simd_width](-(2**14), 0, -1, 2**14)
+    assert_equal(countl_zero(var3), SIMD[int16_t, simd_width](0, 16, 0, 1))
+
+    alias var5 = SIMD[int32_t, simd_width](-(2**30), 0, -1, 2**30)
+    assert_equal(countl_zero(var5), SIMD[int32_t, simd_width](0, 32, 0, 1))
+
+    # BUG: compiler crash after 2**30 for alias, for 2**31
+    #      failed to fold operation pop.cast(#pop<simd 2147483648> : !pop.scalar<index>)
+    # alias var7 = SIMD[int64_t, simd_width](-(2**62), 0, -1, 2**62)
+    alias var7 = SIMD[int64_t, simd_width](
+        -4611686018427387904, 0, -1, 4611686018427387904
+    )
+    assert_equal(countl_zero(var7), SIMD[int64_t, simd_width](0, 64, 0, 1))
+
+
+def test_countr_zero():
+    assert_equal(countr_zero(-(2**59)), 59)
+    assert_equal(countr_zero(-(2**20)), 20)
+    assert_equal(countr_zero(-1), 0)
+    assert_equal(countr_zero(0), 64)
+    assert_equal(countr_zero(1), 0)
+    assert_equal(countr_zero(2), 1)
+    assert_equal(countr_zero(3), 0)
+    assert_equal(countr_zero(4), 2)
+    assert_equal(countr_zero(2**20), 20)
+    assert_equal(countr_zero(2**59), 59)
+
+
+def test_countr_zero_simd():
+    alias simd_width = 4
+    alias int8_t = DType.int8
+    alias int16_t = DType.int16
+    alias int32_t = DType.int32
+    alias int64_t = DType.int64
+
+    alias var1 = SIMD[int8_t, simd_width](-(2**6), 0, -1, 2**6)
+    assert_equal(countr_zero(var1), SIMD[int8_t, simd_width](6, 8, 0, 6))
+
+    alias var3 = SIMD[int16_t, simd_width](-(2**14), 0, -1, 2**14)
+    assert_equal(countr_zero(var3), SIMD[int16_t, simd_width](14, 16, 0, 14))
+
+    alias var5 = SIMD[int32_t, simd_width](-(2**30), 0, -1, 2**30)
+    assert_equal(countr_zero(var5), SIMD[int32_t, simd_width](30, 32, 0, 30))
+
+    # BUG: compiler crash after 2**30 for alias
```

Comment:
@laszlokindrat somehow any number bigger then 2**30(like 2**31, 2**32) causes a compiler crash for aliasing `SIMD[int64]`

Reply:
Could you please file it separately with a minimal reproducer? This probably has something to do with the materialization of `IntLiteral`.